test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java

```diff
@@ -1382,18 +1368,19 @@ public void createUnmanagedExternalIdWillAssign() {
     }
     
     @Test
-    public void cannotUpdateUnmanagedExternalId() {
+    public void canUpdateUnmanagedExternalId() {
         mockHealthCodeAndAccountRetrieval();
+        account.setExternalId(null);
         STUDY.setExternalIdValidationEnabled(false);
-        when(externalIdService.getExternalId(STUDY.getStudyIdentifier(), EXTERNAL_ID)).thenReturn(null);
         
         StudyParticipant participant = new StudyParticipant.Builder().copyOf(PARTICIPANT)
                 .withExternalId("newExternalId").build();
         
         participantService.updateParticipant(STUDY, CALLER_ROLES, participant);
         
-        assertEquals(EXTERNAL_ID, account.getExternalId());
-        verify(externalIdService, never()).assignExternalId(any(), any(), any());
+        assertEquals("newExternalId", account.getExternalId());
+        // This is always called, but it does nothing since there's no record in the table.
+        verify(externalIdService).assignExternalId(any(), eq("newExternalId"), any());
```

Comment:
We should also have test cases for

* deleting external ID
* external ID changes
* same external ID

Reply:
Added.